src/ln/channelmonitor.rs

```diff
@@ -464,8 +467,9 @@ impl ChannelMonitor {
 	/// optional, without it this monitor cannot be used in an SPV client, but you may wish to
 	/// avoid this (or call unset_funding_info) on a monitor you wish to send to a watchtower as it
 	/// provides slightly better privacy.
+	/// It is the responsability of the caller to register outpoint and script with passing the former 
+	/// value as key to update current monitor
 	pub(super) fn set_funding_info(&mut self, funding_info: (OutPoint, Script)) {
-		//TODO: Need to register the given script here with a chain_monitor
```

Comment:
Seems to me that registration of given script is already handled in internal_funding_created or funding_transaction_generated by call to add_update_monitor. If self.funding_txo is set, a call to add_update_monitor_by_key is enough.